include/alpaka/acc/AccOmp5.hpp

```diff
@@ -147,36 +139,45 @@ namespace alpaka
         namespace traits
         {
             //#############################################################################
-            //! The CPU OpenMP 4.0 accelerator accelerator type trait specialization.
+            //! The OpenMP 5.0 accelerator accelerator type trait specialization.
             template<
                 typename TDim,
                 typename TIdx>
             struct AccType<
-                acc::AccCpuOmp4<TDim, TIdx>>
+                acc::AccOmp5<TDim, TIdx>>
             {
-                using type = acc::AccCpuOmp4<TDim, TIdx>;
+                using type = acc::AccOmp5<TDim, TIdx>;
             };
             //#############################################################################
-            //! The CPU OpenMP 4.0 accelerator device properties get trait specialization.
+            //! The OpenMP 5.0 accelerator device properties get trait specialization.
             template<
                 typename TDim,
                 typename TIdx>
             struct GetAccDevProps<
-                acc::AccCpuOmp4<TDim, TIdx>>
+                acc::AccOmp5<TDim, TIdx>>
             {
                 //-----------------------------------------------------------------------------
                 ALPAKA_FN_HOST static auto getAccDevProps(
-                    dev::DevCpu const & dev)
+                    dev::DevOmp5 const & dev)
                 -> acc::AccDevProps<TDim, TIdx>
                 {
+                    alpaka::ignore_unused(dev);
+
+#if defined(ALPAKA_OFFLOAD_MAX_BLOCK_SIZE) && ALPAKA_OFFLOAD_MAX_BLOCK_SIZE>0
+                    auto const blockThreadCount = ALPAKA_OFFLOAD_MAX_BLOCK_SIZE;
+#else
+                    auto const blockThreadCount = ::omp_get_max_threads();
+#endif
 #ifdef ALPAKA_CI
-                    auto const blockThreadCountMax(alpaka::core::clipCast<TIdx>(std::min(4, ::omp_get_max_threads())));
+                    auto const blockThreadCountMax(alpaka::core::clipCast<TIdx>(std::min(4, blockThreadCount)));
+                    auto const gridBlockCountMax(alpaka::core::clipCast<TIdx>(std::min(4, ::omp_get_max_threads())));
 #else
-                    auto const blockThreadCountMax(alpaka::core::clipCast<TIdx>(::omp_get_max_threads()));
+                    auto const blockThreadCountMax(alpaka::core::clipCast<TIdx>(blockThreadCount));
+                    auto const gridBlockCountMax(alpaka::core::clipCast<TIdx>(::omp_get_max_threads())); //! \todo fix max block size for target
```

Comment:
What do you want to do with the TODO here?

Reply:
This is basically a todo for a later OpenMP version: On CPU we would want `blockThreadCountMax * gridBlockCountMax` to equal the number of CPU threads available, so we would write:
```c++
auto const gridBlockCountMax(alpaka::core::clipCast<TIdx>(::omp_get_max_threads() / blockThreadCountMax));
```
However, If we did this, we would always run with one block (if `ALPAKA_OFFLOAD_MAX_BLOCK_SIZE` does not say otherwise) even on GPU, because `omp_get_max_threads()` appears to return the number of threads per block/team when the target is a GPU (at least XL does that, which is the only compiler with which I was actually able to run any of this on a GPU).

What we really want to do is `gridBlockCountMax =  omp_get_max_teams()`, but this function does not exist.

Reply:
The XL behavior may not be conforming to OpenMP 5.0, but I could not test with any other compiler, so any fix for this would at this point be theoretical. The way it is currently handled does not break too much. It can be fixed once it is clear how compilers handle this.

Reply:
Could you add parts of your answer to an extended comment?

Reply:
https://github.com/alpaka-group/alpaka/pull/1126/commits/fbf0df054a673698127df84c73342514bbcb6290